### PR TITLE
#101828 configurable columns for document (file) list 3

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.transaction.annotation.Transactional
+
+open class DocumentenApiColumnDeploymentService(
+    private val documentenApiService: DocumentenApiService,
+) {
+
+    @Transactional
+    @RunWithoutAuthorization
+    @EventListener(DocumentDefinitionDeployedEvent::class)
+    open fun createCaseTabs(event: DocumentDefinitionDeployedEvent) {
```
Comment (marijnritense) [inline zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt]: createCaseTabs?

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if (event.documentDefinition().id().version() == 1L) {
```
Comment (marijnritense) [inline zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt:39]: Shouldn't this only be done if there's no configuration for the columns available? If so, then the `DocumentDefinitionDeployedEvent` might not be the right trigger for this

Comment (Klaas-Ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt:39]: > Shouldn't this only be done if there's no configuration for the columns available?

No. It should only be done when the definition is deployed for the first time

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt]: renamed

File: zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt (hunk 2 above)
Comment (marijnritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt:39]: I should have made that clearer. I meant:

Shouldn't this only be done if there's no configuration for the columns available and it's the deployment is deployed for the first time?

Comment (Klaas-Ritense) [reply zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiColumnDeploymentService.kt:39]: A case can't have configuration if it doesn't exist

**Review by marijnritense — APPROVED** [review]

**Pull request merged**